### ISSUE TITLE
Random exceptions thrown in containers running within ACI

### ISSUE THREAD

**Issue by eugen-nw** [virtual-kubelet/azure-aci#37]
I installed the ACI Connector 7 days ago into an AKS instance and today into another AKS instance.  Both times I used the command below.  I notice right now that my containers fail with AccessViolationException in an unmanaged DLL that we're launching.  The previous deployment work great with the exact same container bits.

`
helm install https://github.com/virtual-kubelet/azure-aci/raw/master/charts/virtual-kubelet-aci-for-aks.tgz --generate-name --set nodeOsType=Windows --set nodeName=virtual-kubelet-aci-connector-for-windows --set image.name=virtual-kubelet --set image.tag=1.2.1 --set providers.azure.vnet.enabled=false --set providers.azure.vnet.subnetName= 
`

I'm no expert at this but running `kubectl describe pod` shows for 7 days ago:
![image](https://user-images.githubusercontent.com/15199814/73037116-aeb25180-3e02-11ea-8931-c1256ee9deed.png)

And for today's installation:
![image](https://user-images.githubusercontent.com/15199814/73037157-d43f5b00-3e02-11ea-9740-304cc8c2e850.png)

If they are indeed different builds, how should I go about installing the previous one that worked?

**Comment by ibabou** [comment]
@eugen-nw do you mean the containers of your pods is the ones that is failing ? VK connector just deploys these pods as CGs to ACI, so VK version won’t have any effect or relation in this case. You can check the CGs under MC_* resource group for your AKS cluster to see more details. What errors do you see ?

**Comment by eugen-nw** [comment]
Thanks very much @ibabou for your prompt response!  You are correct, shortly after I posted this question I switched to installing the containers onto the previous VK Connector and they were failing there as well.

Which brings a more interesting question.  We run a .Net (classic) 4.5.2 Console application in the containers, which loads a C++ Win32 binary to do the calculations.  All failures occur in the Win32 binary, the C# code catches AccessViolationException instances.  Yesterday we've had 7 containers running, some of them were throwing the exception, some were running great and completing the computations successfully.  We've encountered this behavior in the exact same AKS instance on Nov. 22, 2019.  I fixed the problem at that time by manually killing all running containers until the AccessViolationException stopped occurring.  Could it be that ACI is giving us AMD64 containers that can run properly the .Net code but cause the X64 compiled Win32 binary to read from memory locations it should not access?

I'm deploying again the exact same bits that were failing yesterday - we did a rollback in the meantime - I'm curious to see if the exceptions will occur again today.

**Comment by eugen-nw** [comment]
@ibabou I have the "some containers crashing" repro from yesterday going again.  Please let me know if you'd like to investigate it.  As I wrote earlier, the exact same bits crash in one container and run correctly in others.

**Comment by eugen-nw** [comment]
I found what was causing the random behavior.  There were **60** dangling containers in ACI, in addition to the desired count of 7.  All containers were consuming incoming messages from the queue.  The data format changed across the various versions, the Win32 DLL was receiving some uninitialized pieces of data => the AccessViolationException instances.  I updated all ACI Connectors to version 1.2.1 so hopefully the dangling containers will be gone.
